src/main/java/scala_maven/HydraActivateLicenseMojo.java

```diff
@@ -0,0 +1,100 @@
+import org.apache.maven.plugin.MojoFailureException;
+
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.nio.file.Paths;
+
+/**
+ * @goal hydraActivateLicense
+ * @requiresDirectInvocation true
+ */
+public class HydraActivateLicenseMojo extends AbstractMojo {
+
+    /**
+     * @parameter property="hydra.licenseKey"
+     */
+    private String licenseKey;
+
+    /**
+     * @parameter property="hydra.activationServerUrl" default-value="https://activation.triplequote.com/algas/"
+     *
+     */
+    private String activationServerUrl;
+
+    /**
+     * @parameter property="hydra.licenseStore" default-value="${user.home}/.triplequote/hydra.license"
+     *
+     */
+    private String licenseStore;
+
+    /**
+     * The Maven Session Object
+     *
+     * @parameter property="session"
+     * @required
+     * @readonly
+     */
+    protected MavenSession session;
+
+    @Override
+    public void execute() throws MojoExecutionException, MojoFailureException {
+        if (session.getCurrentProject() == session.getTopLevelProject()) {
+            try {
+                activateLicense();
+            } catch (MalformedURLException e) {
+                throw new MojoExecutionException(e, "Failed to parse URL", "");
+            }
+        }
+    }
+
+    private void activateLicense() throws MalformedURLException, MojoExecutionException {
+        getLog().info("Activating license " + licenseKey + " using " + activationServerUrl + ". This may take a while.");
+
+        HydraLicenseStore store = new HydraLicenseStore(
+                new URL(activationServerUrl),
+                Paths.get(licenseStore));
+
+        try {
+            License license = licenseKey == null ? store.activate() : store.activate(licenseKey);
+
+            switch (license.getValidationStatus()) {
+                case LICENSE_VALID:
+                    getLog().debug("License is valid.");
+                    break;
+
+                case LICENSE_INVALID:
+                    throw new MojoExecutionException("License " + licenseKey + " is invalid.");
+
+                case LICENSE_EXPIRED:
+                    throw new MojoExecutionException("License has expired. Please contact " + LicenseMessages.SUPPORT_EMAIL);
+
+                default:
+                    throw new MojoExecutionException("License validity: " + license.getValidationStatus());
+            }
+
+            ActivationStatus activationStatus = license.getActivationStatus();
+
+            switch (activationStatus) {
+                case ACTIVATION_COMPLETED:
+                case DEACTIVATION_COMPLETED:
+                    getLog().info(LicenseMessages.activationMessage(activationStatus));
+                    break;
+
+                default:
+                    throw new MojoExecutionException(LicenseMessages.activationMessage(activationStatus));
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
```

Comment:
I assume this is how it's supposed to be done in a maven plugin, but I'm leaving a comment just in case it's an oversight.